storage/shard.go

```diff
@@ -425,3 +442,17 @@ func (s *dbShard) Flush(ctx context.Context, blockStart time.Time, pm persist.Ma
 
 	return multiErr.FinalError()
 }
+
+func (s *dbShard) CleanupFileset(namespace string, earliestToRetain time.Time) error {
+	filePathPrefix := s.opts.GetCommitLogOptions().GetFilesystemOptions().GetFilePathPrefix()
+	multiErr := xerrors.NewMultiError()
+	expired, err := s.filesetBeforeFn(filePathPrefix, namespace, s.ID(), earliestToRetain)
+	if err != nil {
+		detailedErr := fmt.Errorf("encountered errors when getting fileset files for prefix %s namespace %s shard %d: %v", filePathPrefix, namespace, s.ID(), err)
+		multiErr = multiErr.Add(detailedErr)
+	}
+	if err := s.deleteFilesFn(expired); err != nil {
+		multiErr = multiErr.Add(err)
+	}
+	return multiErr.FinalError()
```

Comment:
Is there a reason we moved CleanupFileset to the shard level?  Seems more clean to have it in the namespace?  Also in the future if we ever do consolidate all shard data into single files we'd have to move this back again.